packages/svelte/src/compiler/migrate/index.js

```diff
@@ -115,9 +115,12 @@ export function migrate(source, { filename } = {}) {
 			derived_components: new Map(),
 			derived_labeled_statements: new Set(),
 			has_svelte_self: false,
-			uses_ts: !!parsed.instance?.attributes.some(
-				(attr) => attr.name === 'lang' && /** @type {any} */ (attr).value[0].data === 'ts'
-			)
+			uses_ts:
+				// Some people could use jsdoc but have a tsconfig.json, so double-check file for jsdoc indicators
+				(use_ts && !source.includes('@type {')) ||
```

Comment:
I think this needs to be more thorough using a regex. It won't pass if you had:

```js
/**
 * @template T
 * @param {T} thing
 * @returns {T}
 */
 ```
 
 For example

Reply:
The problem with these is that I have seen these appear above typescript functions in the wild. `@type` on the other hand is a pretty strong indicator

Reply:
Makes sense. However, if someone specifies `use_ts` to be `true` or `false`, shouldn't that precedence over a check?

Reply:
If they use JSDoc wouldn't they omit the lang="TS"?

Reply:
> If they use JSDoc wouldn't they omit the lang="TS"?

Yes, which is why there's the outer `||` in which case we see "this is definitely TS because the lang tag is set". This is only about the ambiguity of people using JSDoc but having a `tsconfig.json`